File: spec/test_parser_rule_from_multiple_states.c
#include "testparser.h"
#include <assert.h>
#include <string.h>

int main()
{
    char const * input = "a";
    p_context_t * context;
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_UNEXPECTED_TOKEN);
    assert(p_position(context).row == 1);
    assert(p_position(context).col == 2);
    assert(context->token == TOKEN___EOF);
    p_context_delete(context);

    input = "a b";
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    p_context_delete(context);

    input = "bb";
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    p_context_delete(context);

    return 0;
}

File: spec/test_parser_rule_from_multiple_states.rs
use testparser::*;

fn main() {
    let mut context = p_context_new(b"a");
    assert_eq!(P_UNEXPECTED_TOKEN, p_parse(&mut context));
    assert_eq!(1, p_position(&context).row);
    assert_eq!(2, p_position(&context).col);
    assert_eq!(TOKEN___EOF, p_token(&context));
    p_context_delete(context);

    let mut context = p_context_new(b"a b");
    assert_eq!(P_SUCCESS, p_parse(&mut context));
    p_context_delete(context);

    let mut context = p_context_new(b"bb");
    assert_eq!(P_SUCCESS, p_parse(&mut context));
    p_context_delete(context);
}
